x/btcstaking/keeper/grpc_query.go

```diff
@@ -245,6 +229,34 @@ func (k Keeper) LargestBtcReOrg(ctx context.Context, _ *types.QueryLargestBtcReO
 	}, nil
 }
 
+// FinalityProviderBSNId returns the BSN ID for the finality provider with the specified finality provider BTC PK
+func (k Keeper) FinalityProviderBSNId(ctx context.Context, req *types.QueryFinalityProviderBSNIdRequest) (*types.QueryFinalityProviderBSNIdResponse, error) {
```

Comment:
Do we still need this query given that the FP object contains BSN ID as a field?

Reply:
Will remove